lib/capybara/node/finders.rb

```diff
@@ -166,7 +166,13 @@ def all(*args)
       # @return [Capybara::Node::Element]            The found element or nil
       #
       def first(*args)
+        if Capybara.first_default_waiting
+          options = if args.last.is_a?(Hash) then args.pop.dup else {} end
+          args.push({minimum: 1}.merge(options))
+        end
         all(*args).first
+      rescue Capybara::ExpectationNotMet
+        nil
```

Comment:
Actually, v3 is completely another story.

I agree that it would be better if `first` would raise an error if there are no elements rather than return `nil`.